lib/util/git/index.ts

```diff
@@ -597,16 +654,20 @@ export async function mergeBranch(branchName: string): Promise<void> {
   let status: StatusResult;
   try {
     await syncGit();
-    await git.reset(ResetMode.HARD);
-    await git.checkout(['-B', branchName, 'origin/' + branchName]);
-    await git.checkout([
-      '-B',
-      config.currentBranch,
-      'origin/' + config.currentBranch,
-    ]);
-    status = await git.status();
-    await git.merge(['--ff-only', branchName]);
-    await git.push('origin', config.currentBranch);
+    await gitRetry(() => git.reset(ResetMode.HARD));
+    await gitRetry(() =>
+      git.checkout(['-B', branchName, 'origin/' + branchName])
```

Comment:
@rarkins i think we no longer need the first checkout? Looks like it was required before blobless, we now have the full tree offline after clone.

Reply:
OK, but perhaps remove and verify it in a PR before this one?

Reply:
yes, sounds good. @olegkrivtsov Can you create a new PR and remove this first checkout?

Reply:
Sure, I'll do that.

Reply:
I've tried to remove the first checkout, but tests broke: https://github.com/renovatebot/renovate/pull/13509
I'm not sure how to fix this problem. Are you sure this checkout can be removed?